virtual_rainforest/core/base_model.py

```diff
@@ -192,13 +199,61 @@ def setup(self) -> None:
     def spinup(self) -> None:
         """Function to spin up the model."""
 
-    @abstractmethod
     def update(self, time_index: int, **kwargs: Any) -> None:
         """Function to update the model.
 
         Args:
             time_index: The index representing the current time step in the data object.
         """
+        if not self._static:
+            self._update(time_index, **kwargs)
+            return
+
+        self._update_static(time_index, **kwargs)
+
+    @abstractmethod
+    def _update(self, time_index: int, **kwargs: Any) -> None:
+        """Function to update the model.
+
+        Args:
+            time_index: The index representing the current time step in the data object.
+        """
+
+    def _update_static(self, time_index: int, **kwargs: Any) -> None:
+        """Function to update the model in the static case.
+
+        Args:
+            time_index: The index representing the current time step in the data object.
+        """
+        if self._static_data is None:
+            self._update(time_index, **kwargs)
+            self._build_static_data(time_index)
```

Comment:
Not sure that I follow this bit, does the update happen and then new static data gets built for the next time step?

Reply:
Exactly. This is run only once, the first time that we are required to run the update and, thereafter the updated values are used for updating the data object again and again. We need to run update at least once in this case since, otherwise, we do not know what to update the data with. 

If pre-set data is provided as an external file/s, then the `static_data` object is not none and this step is not needed. 

Keeping aside changes in initialisation/validation, the only change to existing models would be to replace `update` by `_update`, since they are overwritting only one of the two options for updating the model. The `_update_static` option is not model specific, and therefore is coded directly in the base model.